client/hooks/AddonAdditionalField/MeinBerlinProcedurePictogram.vue

```diff
@@ -144,6 +163,101 @@ export default {
     getPictogramUrl (hash) {
       return Routing.generate('core_logo', { hash: hash })
     },
+
+    handleDelete () {
+      if (confirm(Translator.trans('check.procedure.pictogram.delete'))) {
+        this.deletePictogram = true
+        this.existingPictogramData = null
+      }
+    },
+
+    async handleUploadSuccess (fileInfo) {
+      try {
+        // First check if the file type is valid
+        const validFormats = ['image/png', 'image/jpeg', 'image/gif']
+        if (fileInfo.type && !validFormats.includes(fileInfo.type)) {
+          dplan.notify.error(Translator.trans('mein.berlin.pictogram.error.format'))
+          dplan.notify.warning(Translator.trans('mein.berlin.pictogram.remove.instruction'))
+          await this.$nextTick()
+          this.removeInvalidFile(fileInfo)
+          return
+        }
+
+        const imageUrl = Routing.generate('core_logo', { hash: fileInfo.hash })
+        const validation = await this.validateImageDimensions(imageUrl, fileInfo.type)
+
+        if (validation.valid) {
+          this.validFile = fileInfo
+          this.deletePictogram = false
+        } else {
+          dplan.notify.error(validation.error)
+          dplan.notify.warning(Translator.trans('mein.berlin.pictogram.remove.instruction'))
+          await this.$nextTick()
+          this.removeInvalidFile(fileInfo)
+        }
+      } catch (error) {
+        console.error('Pictogram validation error:', error)
+        dplan.notify.error(Translator.trans('mein.berlin.pictogram.error.invalid'))
+        await this.$nextTick()
+        this.removeInvalidFile(fileInfo)
+      }
+    },
+
+    removeInvalidFile (fileInfo) {
+      try {
+        // Find the hidden input that stores file hashes for form submission
+        const hiddenInput = document.querySelector('input[name="uploadedFiles[r_pictogram]"]')
+
+        if (hiddenInput) {
+          hiddenInput.value = ''
+        }
+
+        // Remove the visual file display
+        const fileListItems = document.querySelectorAll('[data-cy="uploadFile"] .uploaded-file-item, [data-cy="uploadFile"] li')
```

Comment:
hm, i am not sure, maybe i am missing something, but I do not see the data-cy attributes `uploadFile` or the class `.uploaded-file-item`. What I see in the HTML is `data-cy="uploadFile:uploadedFileItem"`. Why cant we use that directly and need so many selectors?